src/go/rpk/pkg/cli/cluster/partitions/toggle.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+)
+
+func newPartitionEnableCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var partitions []int
+	cmd := &cobra.Command{
+		Use:   "enable [TOPIC]",
```

Comment:
Both enable and disable here only support one topic, whereas `move` (PR #13684) supports many topics (which can also be in the -p flag). Can this PR support the three behaviors?
* topic in arg
* topic in -p
* just partition in -p

Reply:
Also should we have an --all flag for enabling all topics? Is there an easy query we can issue to discover everything disabled, then reenable?

Reply:
Sure, we can support the behaviors, but can I ask what's the benefit of supporting **just** topic in `-p` ? 

it's confusing to me if I should use topic as an arg, or topic in the flag. This will raise questions about if it makes any difference. 

I agree entirely on partition and combinations of topic and partitions (`<namespace>/<topic>/[,partition-id>...]`).

>Is there an easy query we can issue to discover everything disabled, then reenable?

I was expecting the POST for all topics to work this way, but it doesn't, I've asked here: https://github.com/redpanda-data/redpanda/issues/14394#issuecomment-1804953589

Reply:
Sure, we can require a partition in the p flag (but being able to specify topics is nice)

Reply:
> Sure, we can support the behaviors, but can I ask what's the benefit of supporting just topic in -p ?

My 2c. According to the RFC, https://docs.google.com/document/d/1NhM0laLjlwur_57RbSj33q0dstKJf5-RAe_tyShEsmY/edit#heading=h.mvlqwsjv79vf, there's a case the problem is localized to just a few partitions and we want to disable the specific partitions to minimize the impact.

If the problem is localized to topic `test`, the `topic in arg` way helps; `rpk cluster partitions disable test`

If the problem is localized to `test/0` and `test2/1`, the `topic in -p` way helps; `rpk cluster partitions disable -p test/0 -p test2/1`

If the problem is localized to some partitions in topic `test`, the `just partition in -p` way helps; `rpk cluster partitions disable test -p 0,1,2,3,4,5`

Reply:
Yes, that's what I'm adding (expect a force push soon). The point that I didn't quite understand was the ability to **just** select the topic in the flag, i.e:

```
rpk cluster partitions enable --p myTopic
```

But I'm 100% onboard with the need for the other points.